src/components/notifications/HdNotificationsBar.vue

```diff
@@ -41,16 +45,9 @@ export default {
   props: {
     type: {
       type: String,
-      default: 'notification',
+      default: TYPES.NOTIFICATION,
       validator(type) {
-        const allowedTypes = [
-          'notification',
-          'success',
-          'error',
-          'warning',
-        ];
-
-        return allowedTypes.some(allowedType => allowedType === type);
+        return Object.values(TYPES).includes(type);
```

Comment:
I had the same feeling, but it looked to small to be refactored (as a mixin maybe), you'll have to define your types anyway, so the refactoring will affect just this line:
```js
Object.values(TYPES).includes(type);
```
which I think is short enough to be refactored.
Do you already have an idea in mind?

Reply:
I thought about something like:
```javascript
// mixins.js
export default function CreateMixin(MODIFIERS, DEFAULT) {
  return {
    data() {
      return {
        MODIFIERS
      };
    },
    props: {
      modifier: {
        type: String,
        default: DEFAULT,
        validator(modifier) {
          return Object.values(MODIFIERS).includes(modifier);
        }
      }
    }
  };
}
```

And use like:
```javascript
// HelloWorld.vue
import CreateMixin from "@/mixins.js";

const MODIFIERS = {
  PRIMARY: "primary",
  SECONDARY: "secondary"
};

export default {
  name: "HelloWorld",
  mixins: [CreateMixin(MODIFIERS, MODIFIERS.PRIMARY)],
  props: {
    msg: String
  }
};
```

Of course we can play with it a bit, that is a draft, but the idea is to pass `MODIFIERS/TYPES` to the mixin creator and use it.

Thoughts?

Reply:
What about the prop names? You want to keep it always as `modifier`/`type` and `MODIFIERS`/`TYPES` or you want to pass it as a param to the `createMixin` function (which can use named params to avoid confusion)?
So my opinion here is that we will lose code clarity as `MODIFIERS` and `modifier` will be added to the component instance without defining them explicitly (typical mixins disadvantage) and the only advantage I can think of is one or two lines of code less per component.

What do you think @homeday-de/frontend-engineering?

And as you said the first comment, this is out of this PR's scope, but it's nice to discuss it.

Reply:
Ahhh, I think we should keep a default prop. We have type/modifier which will eventually get complicated hahaha
I used modifier in the example but don't focus on the names/structure, just the idea behind it.

Yes, mixins have its drawbacks. It is just an idea as I see us doing it multiple times. It is also out of the scope of this PRs, indeed.

Let's wait for the others and see how it goes.

Reply:
Yeah, I also think if we agree on the naming and the types for all our **multiple types components** then a **Mixin** would be cleaner. otherwise, it won't be that beneficial.